### ISSUE TITLE
Fix plot Y axis label and drop-down menu in Y axis area

### ISSUE THREAD

**Issue by nunoguedelha** [ami-iit/openmct#8]
Refer to https://github.com/ami-iit/openmct/issues/6#issuecomment-1148186551 for the initial analysis.

### Expected Behaviour

- The Y label displays only the value option (e.g. IMU `Roll`).
    <img alt="image" src="https://user-images.githubusercontent.com/6848872/172292943-ef85e7f7-b954-4bed-90a7-766417c61e15.png">
- If in a given range value block of the dictionary we only provide the "unit" field and no "name" field, the respective "unit" is displayed as the Y axis label (The unit is displayed only in the top legend which is an expandable table).
   ```json
   {
     "name": "iCub Telemetry",
     "key": "icubtelemetry",
     "telemetryEntries": [
       {
         "name": "IMU sensor measurements",
         "key": "sens.imu",
         "type": "yarpopenmct.sensormeas",
         "values": [
           {
             "key": "value.ori.roll",
             "unit": "deg",
             "format": "float",
             "hints": {
               "range": 1
             }
           },
           ...
         ]
       },
       ...
     ]
   }
   ```
   ⬇️ 
   <img alt="image" src="https://user-images.githubusercontent.com/6848872/172987287-4129bcb4-ee78-444b-8334-1a51b6d0060f.png">
- The Y axis label has a "vertical-top-bottom-right-to-left" write mode with a 90º clockwise symbol orientation.
   <img width="57" alt="image" src="https://user-images.githubusercontent.com/6848872/173368305-c1543d75-abc0-4d69-b51a-163c5093ee3d.png">

### Observed Issues

- [x] **[I1]** If we only provide the "unit" field and no "name" field, the "unit" isn't displayed either in the Y axis label, leaving only the gear icon.
   <img alt="image" src="https://user-images.githubusercontent.com/6848872/172956332-d71d4852-2f3c-4fda-90ac-b2bc1699f6a4.png">

### Desired Behaviour (delta from baseline)

Beyond the fixed issues, the differences with the expected behaviour shall be:
- [x] **[E1]** The Y axis label has a "vertical-bottom-top-left-to-right" write mode with a 90º counter-clockwise symbol orientation.
   <img width="45" alt="image" src="https://user-images.githubusercontent.com/6848872/173371461-9fc7e79b-e421-4a99-9a4d-43cfc5e8a5af.png">
- [x] **[E2]** The Y label displays the name and/or unit when available (only name defined: `Roll`, only unit defined: `[deg]`, both defined: `Roll [deg]`).
   | Pitch | Roll |
   | --- | --- |
   | <img alt="image" src="https://user-images.githubusercontent.com/6848872/173374534-cbc3fe20-3db3-4686-bbc2-29dfe3e3328e.png"> | <img alt="image" src="https://user-images.githubusercontent.com/6848872/173374722-f57fe2cd-e4db-43fc-95f5-b16b8c5aa317.png"> |

Fixed in #7 .

**Comment by nunoguedelha** [comment]
## Background: the `updateFromSeries` method

This method updates the `yAxis` format, values, and label from the plot "series" object. It sets the axis label
- either from the "range" block `name` attribute if it is defined
   https://github.com/ami-iit/openmct/blob/7e67bdba9ff7d6c4a98fa952ce484b3d0f474584/src/plugins/plot/configuration/YAxisModel.js#L200-L204
- either from the "range" block `units` attribute if it is defined
   https://github.com/ami-iit/openmct/blob/7e67bdba9ff7d6c4a98fa952ce484b3d0f474584/src/plugins/plot/configuration/YAxisModel.js#L220-L224

There are two bugs arising here:
1. **bug:** The "range" attribute `units` used to update `labelUnits` is not aligned with the equivalent attribute used elsewhere in the code:
   ```
   $ egrep -Rs -e "\.unit[^s]" src/plugins/*
   src/plugins/LADTable/components/LADTable.vue:            let metadataWithUnits = valueMetadatas.filter(metadatum => metadatum.unit);
   src/plugins/LADTable/components/LadTableSet.vue:                        if (metadatum.unit) {
   src/plugins/LADTable/components/LADRow.vue:            this.unit = this.valueMetadata.unit || '';
   src/plugins/displayLayout/DisplayLayoutToolbar.js:                            .filter((metadatum) => metadatum.unit)
   src/plugins/displayLayout/components/TelemetryView.vue:            let unit = this.metadata.value(value).unit;
   src/plugins/displayLayout/components/TelemetryView.vue:            const unit = this.unit ? ` ${this.unit}` : '';
   src/plugins/plot/legend/PlotLegendItemExpanded.vue:            this.unit = seriesObject.get('unit');
   src/plugins/plot/configuration/PlotSeries.js:            unit: range.unit,
   src/plugins/telemetryTable/TelemetryTable.js:                if (metadatum.unit !== undefined) {
   src/plugins/telemetryTable/components/table-configuration.vue:                if (metadatum.unit !== undefined) {
   src/plugins/telemetryTable/TelemetryTableUnitColumn.js:                    return this.metadatum.unit;
   src/plugins/telemetryTable/TelemetryTableUnitColumn.js:                    return this.metadatum.unit;
   src/plugins/telemetryTable/TelemetryTableUnitColumn.js:            return this.metadatum.unit;
   ```
   **This results in the `labelUnits` being undefined although we specify it in the dictionary.**
2.  **bug:** Whatever the resulting Y axis `label` processed here, that label is overwritten in `toggleYAxisLabel` from `thist.yAxisLabel`:
   https://github.com/ami-iit/openmct/blob/7e67bdba9ff7d6c4a98fa952ce484b3d0f474584/src/plugins/plot/axis/YAxis.vue#L129
   ➡️ These result in the observed issue **[I1]** in the initial description.

### Note
We cannot have a Y axis format like `<labelName> [<labelUnits>]`, like or instance `Pitch [deg]` because the implemented choice between `labelUnits` and `labelName` is exclusive.
   ➡️ This needs t be changed for unlocking enhancement **[E2]**.

## Fixes
- [x] 1 ➡️ Replace `units` by `unit` wherever needed. This is interdependent with https://github.com/ami-iit/yarp-openmct/issues/119.
- [x] 2 ➡️ Read from `this.yAxis.model.label` instead of writing to: `this.yAxisLabel = this.yAxis.get...`.

_Originally posted by @nunoguedelha in https://github.com/ami-iit/openmct/issues/6#issuecomment-1151350781_

**Comment by nunoguedelha** [comment]
## Replace `units` by `unit` wherever relevant in the plugins

```
$ egrep -Rs -e "units" src/plugins/*
```
- [x] ~src/plugins/LADTable/components/LADTable.vue:                :has-units="hasUnits"~
- [x] ~src/plugins/LADTable/components/LadTableSet.vue:                :has-units="hasUnits"~
- [x] ~src/plugins/LADTable/components/LADRow.vue:        class="js-units"~
- [x] src/plugins/charts/BarGraphPlot.vue:                const unit = yAxisMetadata.units;
- [x] src/plugins/charts/BarGraphView.vue:            const xAxisUnit = xAxisMetadata.units ? `(${xAxisMetadata.units})` : '';
- [x] src/plugins/charts/BarGraphView.vue:            const yAxisUnit = yAxisMetadata.units ? `(${yAxisMetadata.units})` : '';
- [x] ~src/plugins/displayLayout/DisplayLayoutToolbar.js:                    applicableItems = unitsOnly(applicableItems);~
- [x] ~src/plugins/displayLayout/DisplayLayoutToolbar.js:                                title: "Show units"~
- [x] ~src/plugins/displayLayout/DisplayLayoutToolbar.js:                                title: "Hide units"~
- [x] ~src/plugins/displayLayout/DisplayLayoutToolbar.js:                function unitsOnly(items) {~
- [x] src/plugins/plot/configuration/YAxisModel.js:                return s.metadata ? s.metadata.value(s.get('yKey')).units : '';
- [x] src/plugins/plot/inspector/forms/LegendForm.vue:                <option value="units">Units</option>
- [x] ~src/plugins/telemetryTable/TelemetryTable.js:                // add units column if available~
- [x] ~src/plugins/telemetryTable/components/table-configuration.vue:                // if units are available, need to add columns to be hidden~

_Originally posted by @nunoguedelha in https://github.com/ami-iit/openmct/issues/6#issuecomment-1151550038_

**Comment by nunoguedelha** [comment]
## Read from this.yAxis.model.label instead of writing to

If we remove the attribute "name" (="Roll") from the first range value of the iCub Telemetry dictionary entries...
```json
{
  "name": "iCub Telemetry",
  "key": "icubtelemetry",
  "telemetryEntries": [
    {
      "name": "IMU sensor measurements",
      "key": "sens.imu",
      "type": "yarpopenmct.sensormeas",
      "values": [
        {
          "key": "value.ori.roll",
          "unit": "deg",
          "format": "float",
          "hints": {
            "range": 1
          }
        },
        ...
      ]
    },
    ...
  ]
}
```
We should see "deg" displayed in the plot Y axis label instead of "Roll" or no label, but we actually see NO label, while the Y axis model object label `yAxis.model.label` is set to `'deg'`.

<img width="770" alt="image" src="https://user-images.githubusercontent.com/6848872/172956332-d71d4852-2f3c-4fda-90ac-b2bc1699f6a4.png">

If we update the attribute `this.yAxisLabel` as follows
```js
        toggleYAxisLabel() {
            let yAxisObject = this.yKeyOptions.filter(o => o.name === this.yAxisLabel)[0];

            if (yAxisObject) {
                this.$emit('yKeyChanged', yAxisObject.key);
                this.yAxisLabel = this.yAxis.get('label');
            }
        }
```
at initialisation we still see NO label, and this is due to line 
https://github.com/ami-iit/openmct/blob/7e67bdba9ff7d6c4a98fa952ce484b3d0f474584/src/plugins/plot/axis/YAxis.vue#L117
which overwrites `this.yAxisLabel` to the first name among the available options (empty in our case), whatever the result produced by `updateFromSeries`.

Nevertheless, after switch the field option a couple of times coming back to the option "Roll", we get the expected axis label `deg` (the unit).

<img width="693" alt="image" src="https://user-images.githubusercontent.com/6848872/172987287-4129bcb4-ee78-444b-8334-1a51b6d0060f.png">

We can solve the initialisation problem by updating `this.yAxisLabel` as soon as `this.yAxis` is initialised in
https://github.com/ami-iit/openmct/blob/7e67bdba9ff7d6c4a98fa952ce484b3d0f474584/src/plugins/plot/axis/YAxis.vue#L93-L97

⬇️ 

```js
    mounted() {
        this.yAxis = this.getYAxisFromConfig();
        this.yAxisLabel = this.yAxis.get('label');
        this.loaded = true;
        this.setUpYAxisOptions();
    },
```

_Originally posted by @nunoguedelha in https://github.com/ami-iit/openmct/issues/6#issuecomment-1151910663_

**Comment by nunoguedelha** [comment]
## Desired Behaviour [E1]: Y axis label text format

We wish the Y axis label text to appear as follows:

<img width="45" alt="image" src="https://user-images.githubusercontent.com/6848872/173371461-9fc7e79b-e421-4a99-9a4d-43cfc5e8a5af.png">

This can be obtained by applying:
- a `sideways-lr` _writing mode_: 
  For ltr scripts, content flows vertically from bottom to top. All the glyphs, even those in vertical scripts, are set sideways toward the left. :warning: This is supported only on Firefox. :warning:
- a `mixed` _text-orientation_:
  _text-orientation_ sets the orientation of the text characters in a line (supported on all browsers except Internet Explorer). `mixed` orientation rotates the characters of horizontal scripts 90° counter-clockwise. It lays out the characters of vertical scripts naturally. Default value.
- a `ltr` _direction_. Supported on all browsers.

| Enough vertical height | Vertical height too short |
| --- | --- |
| <img alt="image" src="https://user-images.githubusercontent.com/6848872/173492874-aca20f94-b9df-4b9f-ad3e-675331166a64.png"> | <img alt="image" src="https://user-images.githubusercontent.com/6848872/173492933-3ae82b68-e067-44be-894b-c6a838dc630e.png"> |


Because of `sideways-lr` _writing mode_ to be supported only on Firefox, we need another approach, e.g. the CSS transformation "rotate": `rotate(180deg)`

References:
- Scripts, languages and script directionality: https://www.w3.org/International/questions/qa-scripts.en
- `writing-mode` : https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode
- `text-orientation` : https://developer.mozilla.org/en-US/docs/Web/CSS/text-orientation
- `direction` : https://developer.mozilla.org/en-US/docs/Web/CSS/direction

**Comment by nunoguedelha** [comment]
## Desired Behaviour [E2]: The Y label displays the name and/or unit when available

...

**Comment by nunoguedelha** [comment]
Fixed in #7 .
